orttraining/orttraining/python/training/ortmodule/torch_cpp_extensions/__init__.py

```diff
@@ -0,0 +1,45 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+"""Support for PyTorch C++ extensions within ORTModule
+
+
+TODO: Implement mechanism to register extensions and prevent issues with incorrect/missing flags
+      for each :meth:`torch.utils.cpp_extension.*` call
+"""
+
+import threading
+from functools import wraps
+from onnxruntime.capi import _pybind_state as C
+
+
+def run_once_aten_op_executor(f):
+    """
+    Decorator to run a function only once.
+    :param f: function to be run only once during execution time despite the number of calls
+    :return: The original function with the params passed to it if it hasn't already been run before
+    """
+    @wraps(f)
+    def aten_op_executor_wrapper(*args, **kwargs):
+        if not aten_op_executor_wrapper.has_run:
+            with aten_op_executor_wrapper.lock:
+                if not aten_op_executor_wrapper.has_run:
+                    aten_op_executor_wrapper.has_run = True
+                    return f(*args, **kwargs)
+
+    aten_op_executor_wrapper.lock = threading.Lock()
+    aten_op_executor_wrapper.has_run = False
```

Comment:
If the intention is to run the method ```aten_op_executor_wrapper```  just once by using the ```aten_op_executor_wrapper.has_run``` flag, won't resetting it here defeat the purpose?

Reply:
This code is unchanged from @iK1D